packages/components/src/text-control/index.tsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import type { ChangeEvent, ForwardedRef } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import BaseControl from '../base-control';
+import type { WordPressComponentProps } from '../ui/context';
+import type { TextControlProps } from './types';
+
+function UnforwardedTextControl(
```

Comment:
By the way, sorry about ef3a12e! I fixed up the contributing guide in #40682.

Reply:
No worries :) Great that you updated the guide. Did follow the guide and seems to cover most of the stuff you need.

Reply:
Oooh, good to hear! 🙌

Reply:
Thank you for the feedback! And of course do let us know if you feel like there's something missing (or something wrong) in the guide :)